samcli/local/docker/lambda_image.py

```diff
@@ -287,3 +289,18 @@ def _generate_dockerfile(base_image, layers):
         for layer in layers:
             dockerfile_content = dockerfile_content + f"ADD {layer.name} {LambdaImage._LAYERS_DIR}\n"
         return dockerfile_content
+
+    def _remove_images(self, repo_name: str) -> None:
+        """
+        Remove all images for given repo
+
+        Parameters
+        ----------
+        repo_name str
+            Repo for which all images will be removed
+        """
+        try:
+            for image in self.docker_client.images.list(name=repo_name):
+                self.docker_client.images.remove(image.id)
+        except docker.errors.APIError:
+            LOG.exception("Failed to remove images")
```

Comment:
I wasn't sure if I should raise a user exception here, because I don't think the building of the new image should fail if deleting old images fail. So this would just add a log message, and the old images would be deleted (or at least tried to be deleted again) the next time a new image is built.